tripy/tripy/backend/compiler_api.py

```diff
@@ -33,6 +33,130 @@
+
+    Usage:
+        default = default_stream()  # Returns the default Stream object for the current device
+        default_gpu1 = default_stream(1)  # Returns the default Stream object for GPU 1
+
+    Note:
+        Calling default_stream() with the same device always returns the same Stream instance for that device.
+    """
+
+    _instances = {}
+
+    def __new__(cls, device=None):
+        if device is None:
+            device = 0
+
+        if device != 0:
+            raise_error(f"Tripy stream only works with device index 0, got device={device}")
+
+        if device not in cls._instances:
+            instance = super().__new__(cls)
+            instance.stream = Stream()
+            cls._instances[device] = instance.stream
+
+        return cls._instances[device]
+
+
+@export.public_api(document_under="compiler")
+def synchronize():
+    """
+    Synchronizes the default stream.
+
+    .. code-block:: python
+        :linenos:
+        :caption: Use streams in execution
+
+        linear = tp.Linear(2, 3)
+        compiler = tp.Compiler(linear)
+
+        compiled_linear = compiler.compile(tp.InputInfo((2, 2), dtype=tp.float32))
+
+        a = tp.ones((2, 2), dtype=tp.float32)
+
+        out = compiled_linear(a)
+        tp.synchronize()
+        assert cp.array_equal(cp.from_dlpack(out), cp.from_dlpack(linear(a)))
+    """
+    default_stream().synchronize()
+
+
+@export.public_api(document_under="compiler")
+class Stream:
+    """
+    Represents a CUDA stream that can be used to manage concurrent operations.
+
+    This class is a wrapper around the underlying stream object, allowing management of CUDA streams.
+
+    Args:
+        priority (int): Assign priority for the new stream. Lower number signifies higher priority.
+
+    .. code-block:: python
+        :linenos:
+        :caption: Creating new streams
+
+        streamA = tp.Stream()
+        streamB = tp.Stream()
+        defaultStream = tp.default_stream()
+        assert streamA != streamB
+        assert defaultStream != streamB
+
+    .. code-block:: python
+        :linenos:
+        :caption: Example
+
+        linear = tp.Linear(2, 3)
+        compiler = tp.Compiler(linear)
+
+        compiled_linear = compiler.compile(tp.InputInfo((2, 2), dtype=tp.float32))
+
+        a = tp.ones((2, 2), dtype=tp.float32)
+
+        compiled_linear.stream = tp.Stream()
+        out = compiled_linear(a)
+        assert cp.array_equal(cp.from_dlpack(out), cp.from_dlpack(linear(a)))
+    """
+
+    def __init__(self, priority=0):
+        if priority != 0:
+            raise_error(
+                "Incorrect stream priority",
+                [f"Stream priority can only be 0 until #172 is fixed, got priority={priority}."],
+            )
+        from tripy.backend.mlir.utils import MLIRRuntimeClient
+
+        self._active_cuda_stream = MLIRRuntimeClient().create_stream()
+
+    def synchronize(self):
```

Comment:
Needs return type. Please check docs can be built successfully.

Reply:
Fixed.